### DISCUSSION TITLE
Automatic color contrast with CSS Variables

### DISCUSSION THREAD

**Discussion by Bartmr** [twbs/bootstrap#36043] [Ideas]
I propose a way [here](https://github.com/twbs/bootstrap/issues/36042) of how we can have automatic text/elements contrast, and stop having to hardcode color utility classes in our pages.

This can also be helpful in the future for having dark mode.

I already use this system daily, and it went into production on some products.

**Comment by WinterSilence** [comment]
SCSS variables is "hardcode color"? can you show some products where you use so many colors?
